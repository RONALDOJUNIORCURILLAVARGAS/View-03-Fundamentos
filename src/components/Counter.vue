<template>
<div>
    <h2>{{ customTitle}}</h2>
    <p> {{counter}}<sup>2</sup>={{SquareCounter}} </p>
    <div>
        <button @click="increase">+1</button>
        <button @click="decrease">-1</button>
    </div>

</div>
  
</template>

<script>
export default {
    props:{
        title:String,
        start:{
            type:Number,
            default:100,
            validator(value){
                return value>=100 
            }
           // required:true
        }
    },
    data() {
        return {
            counter:this.start
        }
    },
    methods: {
        getSquareValue(){
            //console.log('ejecucion del get');
            return this.counter * this.counter;
        },
        increase(){
            ++this.counter;

        },
        decrease(){
            --this.counter;
        }
        //decrease
    },
    //Las propieadades computadas se guardan en el cache los methods no
    computed:{
        SquareCounter(){
            //console.log('Computed Square counter')
            return this.counter * this.counter;
        },
        customTitle(){
           // return this.title?this.title:'Counter';
            return this.title || 'Counter';
        }
    }

}
</script>

<style>
button{
    background-color: #64BB87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;

}
button:hover{
    background-color: #5aa67b;
    transition: 0.3s ease-in-out;
}
</style>
