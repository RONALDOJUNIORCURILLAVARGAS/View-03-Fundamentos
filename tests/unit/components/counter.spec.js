import {shallowMount,mount} from '@vue/test-utils'
import Counter from '@/components/Counter'
describe('Counter component',()=>{
//Test para capturar el html y verificar su modificacion
//En el caso de actualizar la prueba y el codigo haya sido modificado ejecutar 
// el comando: yarn test:unit -u      
//Que actualizara el html almacenado con update(-u)

    /* test('debe de hacer match con el snapchot',()=>{
      
        const wrapper= shallowMount(Counter)
        
        expect(wrapper.html()).toMatchSnapshot()  
    }) */
    test('h2 debe de tener el valor por defecto "Counter" ', () => {
        const wrapper= shallowMount(Counter)
        //Verificamos que exista 
        //y que el valor sea true cuando exista (toBeTruthy)
        //o puedes escribir toBe(true) 
        expect(wrapper.find('h2').exists()).toBeTruthy()
        //El findALl captura muchos parrafos q referencien al argumento q se busque
        //En el uso de find solo me trae al primero q encuentre

        const h2=wrapper.find('h2')
        console.log(h2.text())
        //Comparamos que lo que recibe es Counter dentro de la etiqueta de h2
        expect(h2.text()).toBe('Counter')
    })
    
});
