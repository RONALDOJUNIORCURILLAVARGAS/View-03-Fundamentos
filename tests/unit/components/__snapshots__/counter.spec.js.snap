// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component debe de hacer match con el snapchot 1`] = `
<div>
  <h2>Counter</h2>
  <p>100<sup>2</sup>=10000</p>
  <div class="button"><button>+1</button><button>-1</button></div>
</div>
`;
